src/main/java/edu/cornell/kfs/ksr/document/SecurityGroupMaintainable.java

```diff
@@ -41,5 +47,28 @@ public void processAfterCopy(MaintenanceDocument document, Map<String, String[]>
             }
         }
     }
+    
+    @Override
+    public void doRouteStatusChange(DocumentHeader documentHeader) {
+        super.doRouteStatusChange(documentHeader);
+
+        SecurityGroup securityGroup = (SecurityGroup) getDataObject();
+        createSecurityProvisioningEntryIfItDoesNotExist(securityGroup);
+    }
+    
+    private void createSecurityProvisioningEntryIfItDoesNotExist(SecurityGroup securityGroup) {
+        Map<String, Object> hashMap = new HashMap<String, Object>();
+        hashMap.put(KSRConstants.SECURITY_GROUP_ID, securityGroup.getSecurityGroupId());
+        BusinessObjectService businessObjectService = SpringContext.getBean(BusinessObjectService.class);
```

Comment:
One of the superclasses should already have a getBusinessObjectService() helper method, so you don't need to manually invoke SpringContext here.

Reply:
updated